src/devices/machine/smc91c9x.cpp

```diff
@@ -140,14 +137,42 @@ void smc91c9x_device::device_start()
 
 void smc91c9x_device::device_reset()
 {
+	unsigned char i;
+	const unsigned char * mac;
+
 	memset(m_reg, 0, sizeof(m_reg));
+
+	memset(m_rx, 0, sizeof(m_rx));
+	memset(m_tx, 0, sizeof(m_tx));
+
 	memset(m_regmask, 0, sizeof(m_regmask));
 	m_irq_state = 0;
 	m_alloc_count = 0;
-	m_fifo_count = 0;
+	rx_fifo_out = 0;
+	rx_fifo_in = 0;
+
+	tx_fifo_out = 0;
+	tx_fifo_in = 0;
+
 	m_sent = 0;
 	m_recd = 0;
 
+	osd_list_network_adapters();
+
+	mac = (const unsigned char *)get_mac();
+
+	if (LOG_ETHERNET)
+	{
+		logerror("MAC : ");
+		for(i=0;i<ETHERNET_ADDR_SIZE;i++)
+		{
+			logerror("%.2X",mac[i]);
+		}
+		logerror("\r\n");
```

Comment:
Seriously?  The `logerror` output will be in text mode, so a line should be ended with "\n", not "\r\n" - line endings will be translated as necessary.  This applies to all the other added logging, too.

Reply:
Done in the new PR. I wasn't using logerror and the \r\n was needed with the output i used. I just forgot to revert to \n before the pull request.